### PR TITLE
Split internal/ into examples/ and scripts/

### PR DESCRIPTION
**Pull request by abitrolly**
## What type of PR is this?

- cleanup

## What this PR does / why we need it:

Move code into dirs that explain the purpose.

## Which issue(s) this PR fixes:

https://github.com/urfave/cli/pull/1991#issuecomment-2434179786

## Testing

_(fill-in or delete this section)_

## Release Notes

_(REQUIRED)_

```release-note

```

### CODE REVIEW

File: .github/codecov.yml [hunk 1]
```diff
@@ -8,4 +8,5 @@ coverage:
       default:
         threshold: 5%
 ignore:
-  - internal
+  - examples
+  - scripts
```
Comment (dearchap) [inline .github/codecov.yml:12]: just wondering why not just call it build instead of scripts ?

Comment (abitrolly) [reply .github/codecov.yml:12]: I remember some C and Python projects used build/ for target directory with artifacts and temp files.

And dir like scripts/ is a part of a wider convention for project automation https://github.blog/engineering/scripts-to-rule-them-all/

Comment (Juneezee) [reply .github/codecov.yml:12]: `examples/` and `scripts/` are also pretty common in Go projects.

Though not official, the https://github.com/golang-standards/project-layout recommends this pattern.

**Review by dearchap — APPROVED** [review]

**Comment by dearchap** [comment]
@abitrolly Thanks for the PR !!!

**Pull request merged**